maintenance/uploadXmlDumpToSpectrumS3Bucket.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+require_once __DIR__ . '/../Maintenance.php';
+
+/**
+ * PLATFORM-4161: uploads XML dumps for public wikis to S3 bucket.
+ *
+ * Script need to be executed with 2 arguments: AWSAccessKey, AWSSecretKey and bucket name.
+ *
+ * use --saveChanges to actually prepare and save XML dumps
+ */
+
+class UploadXmlDumpToSpectrumS3Bucket extends Maintenance {
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = 'Downloads XML dump for given wiki and uploads it to Spectrum S3 bucket';
+		$this->addOption( 'AWSAccessKey', 'Spectrum AWS access key', true, true, 'a' );
+		$this->addOption( 'AWSSecretKey', 'Spectrum AWS secret key', true, true, 's' );
+		$this->addOption( 'bucketName', 'S3 bucket name', true, true, 'b' );
+		$this->addOption( 'saveChanges', 'Get XML dump and save it in s3 bucket', false, false, 'd' );
+	}
+
+	public function execute() {
+
+		global $wgMaxShellTime, $wgMaxShellFileSize, $wgDumpsDisabledWikis;
+
+		$AWSAccessKey = $this->getOption( 'AWSAccessKey' );
+		$AWSSecretKey = $this->getOption( 'AWSSecretKey' );
+		$bucketName =  $this->getOption( 'bucketName' );
+		$saveChanges = $this->hasOption( 'saveChanges' );
+
+		$range = array();
+		$range[] = "city_public = 1";
+
+		if( !empty( $wgDumpsDisabledWikis ) && is_array( $wgDumpsDisabledWikis ) ) {
+			$range[] = 'city_id NOT IN (' . implode( ',', $wgDumpsDisabledWikis ) . ')';
+		}
+
+		$wgMaxShellTime = 0;
+		$wgMaxShellFileSize = 0;
+
+		$dbw = Wikifactory::db( DB_MASTER );
+		$sth = $dbw->select(
+			array( "city_list" ),
+			array( "city_id", "city_dbname" ),
+			$range,
+			__METHOD__,
+			array( "ORDER BY" => "city_id" )
+		);
+
+		$s3 = new S3( $AWSAccessKey, $AWSSecretKey );
+
+		while( $row = $dbw->fetchObject( $sth ) ) {
```

Comment:
Just wondering now. It seems that most of maintenance scripts are implement to handle a single wiki at a time and if there is a need to run a script against all of them it's being done outside of the mechanism. 

I think that we could stick to this convention and implement a separate script that iterates over city_list and manages progress, failures from single wikis and so on, but it might be difficult

Reply:
yeah, we could do it this way, but do we want to create another script that would be used only to run this one?

Reply:
@Michal-Chatlas1221 It is due to the historical reasons. We used to have `run_maintenace` script that invoked given script in context of multiple wiki's. We do have plans to re-introduce the mechanism in some way but we need that to be compatible with our stack (k8s).